eth/beacon/types/states.py

```diff
@@ -76,27 +83,27 @@ class BeaconState(rlp.Serializable):
 
     def __init__(
             self,
-            slot: int,
-            genesis_time: int,
+            slot: SlotNumber,
+            genesis_time: Timestamp,
             fork_data: ForkData,
-            validator_registry_latest_change_slot: int,
+            validator_registry_latest_change_slot: SlotNumber,
             validator_registry_exit_count: int,
             validator_registry_delta_chain_tip: Hash32,
-            previous_justified_slot: int,
-            justified_slot: int,
-            justification_bitfield: int,
-            finalized_slot: int,
+            previous_justified_slot: SlotNumber,
+            justified_slot: SlotNumber,
+            justification_bitfield: Bitfield,
```

Comment:
Maybe leave a comment: "TODO: check if `justification_bitfield` is `bytes` or `int`".